src/main/java/me/gallowsdove/foxymachines/listeners/GhostBlockListener.java

```diff
@@ -0,0 +1,21 @@
+package me.gallowsdove.foxymachines.listeners;
+
+import me.gallowsdove.foxymachines.implementation.materials.GhostBlock;
+import org.bukkit.entity.FallingBlock;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerFishEvent;
+import org.bukkit.persistence.PersistentDataType;
+
+public class GhostBlockListener implements Listener {
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    private void onHitByFishingRod(PlayerFishEvent e) {
+        if (!(e.getCaught() instanceof FallingBlock b)) {
+            return;
+        }
+        if (b.getPersistentDataContainer().has(GhostBlock.KEY, PersistentDataType.STRING)) {
+            e.setCancelled(true);
+        }
```

Comment:
Maybe just merge this into single if statement? I find it weird to use return before just a single if.